velocity/src/main/java/me/leoko/advancedban/velocity/VelocityMethods.java

```diff
@@ -0,0 +1,377 @@
+
+  @Override
+  public void setCommandExecutor(String cmd, TabCompleter tabCompleter) {
+    CommandMeta meta = server.getCommandManager().metaBuilder(cmd).build();
+    server.getCommandManager().register(meta, new CommandReceiverVelocity(server, cmd));
+  }
+
+  @Override
+  public void sendMessage(Object player, String msg) {
+      ((CommandSource) player).sendMessage(LegacyComponentSerializer.legacyAmpersand().deserialize(msg));;
+  }
+
+  @Override
+  public String getName(Object player) {
+    if (player instanceof Player) {
+      return ((Player) player).getUsername();
+    } else {
+      return "CONSOLE";
+    }
+  }
+
+  @Override
+  public String getName(String uuid) {
+      return server.getPlayer(UUID.fromString(uuid)).get().getUsername();
+  }
+
+  @Override
+  public String getIP(Object player) {
+    return ((Player)player).getRemoteAddress().getAddress().getHostAddress();
+  }
+
+  @Override
+  public String getInternUUID(Object player) {
+    return player instanceof  Player ? ((Player) player).getUniqueId().toString().replace("-", "") : "none";
+  }
+
+  @Override
+  public String getInternUUID(String player) {
+    Optional<Player> optionalPlayer = server.getPlayer(player);
+    return optionalPlayer.map(value -> value.getUniqueId().toString().replace("-", "")).orElse(null);
+  }
+
+  @Override
+  public boolean hasPerms(Object player, String perms) {
+      return ((CommandSource) player).hasPermission(perms);
+  }
+
+  @Override
+  public Permissionable getOfflinePermissionPlayer(String name) {
+    if(luckPermsSupport) return new LuckPermsOfflineUser(name);
+
+    return permission -> false;
+  }
+
+  @Override
+  public boolean isOnline(String name) {
+    return server.getPlayer(name).isPresent();
+  }
+
+  @Override
+  public Player getPlayer(String name) {
+    if(server.getPlayer(name).isPresent()) {
+      return server.getPlayer(name).orElse(null);
+    }
+    return null;
+  }
+
+  @Override
+  public void kickPlayer(String player, String reason) {
+    server.getPlayer(player).get().disconnect(LegacyComponentSerializer.legacyAmpersand().deserialize(reason));
+  }
+
+  @Override
+  public Player[] getOnlinePlayers() {
+    return server.getAllPlayers().toArray(new Player[]{});
+  }
+
+  @Override
+  public void scheduleAsyncRep(Runnable rn, long l1, long l2) {
+    server.getScheduler().buildTask(getPlugin(), rn).delay(l1*50, TimeUnit.MILLISECONDS).repeat(l2*50, TimeUnit.MILLISECONDS);
+  }
+
+  @Override
+  public void scheduleAsync(Runnable rn, long l1) {
+    server.getScheduler().buildTask(getPlugin(), rn).delay(l1*50, TimeUnit.MILLISECONDS).schedule();
+  }
+
+  @Override
+  public void runAsync(Runnable rn) {
+    server.getScheduler().buildTask(getPlugin(), rn).schedule();
+  }
+
+  @Override
+  public void runSync(Runnable rn) {
+    rn.run();
+  }
+
+  @Override
+  public void executeCommand(String cmd) {
+    server.getCommandManager().executeImmediatelyAsync(server.getConsoleCommandSource(), cmd);
```

Comment:
Two things here:
1. This should probably use `executeAsync` rather than `executeImmediatelyAsync`, as I don't see a reason to skip the CommandExecuteEvent.

2. Per CompletableFuture best practices, an exception handler should be attached to the returned CompletableFuture, since this future is neither chained nor returned nor awaited:
```java
methodReturningFuture().exceptionally((ex) -> {
  ex.printStackTrace();
  return null;
});
```
executeImmediatelyAsync and executeAsync both return CompletableFuture